env/env_posix.cc

```diff
@@ -519,9 +520,12 @@ Env* Env::Default() {
   CompressionContextCache::InitSingleton();
   INIT_SYNC_POINT_SINGLETONS();
   static PosixEnv default_env;
-  static CompositeEnvWrapper composite_env(&default_env,
-                                           FileSystem::Default().get());
-  return &composite_env;
+  return &default_env;
+}
+
+std::unique_ptr<Env> NewCompositeEnv(std::shared_ptr<FileSystem> fs) {
+  PosixEnv* default_env = static_cast<PosixEnv*>(Env::Default());
+  return std::unique_ptr<Env>(new PosixEnv(default_env, fs));
 }
```

Comment:
This seems like the wrong approach to me.  Can you explain why we need this method at all?

Reply:
See previous comment about backward compatibility.

Reply:
This logic is already accomplished through the CompositeEnv class and doesn't need to be pushed into the PosixEnv one.

I think what you are after is that there should be an Env::Default(std::shared_ptr<FileSystem>) API added.  This way, you could create a "PosixEnv" using a non-Posix File system.  Additionally, there should be a std::shared_ptr<FileSystem> & FileSystem::Default() call that returns a PosixFileSystem.

Reply:
The FileSystem::Default() API already exists. NewCompositeEnv(std::shared_ptr<FileSystem>) is basically the Env::Default(std::shared_ptr) API that you're referring to.